build failure if compiled without MBEDTLS_PK_RSA_ALT_SUPPORT

Note: This is just a template, so feel free to use/remove the unnecessary things

### Description
- Type: Bug
- Priority: Major

---------------------------------------------------------------
## Bug

**OS**  
linux

**mbed TLS build:**  
Version: 4500e83fe41a0765bc06b47662b3ae00d0b1bd59
OS version: 5.6.0
Configuration: please attach config.h file where possible  
Compiler and options (if you used a pre-built binary, please indicate how you obtained it):  
Additional environment information:  

**Peer device TLS stack and version**  
n/a
Version:  

**Expected behavior**   
builds without errors

**Actual behavior**  
[ 74%] Linking C executable test_suite_x509write
/usr/lib64/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: CMakeFiles/test_suite_x509write.dir/test_suite_x509write.c.o: in function `execute_tests':
test_suite_x509write.c:(.text+0x1756): warning: warning: your code uses stdio (7+k bloat).
/usr/lib64/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: /opt/diet/lib-x86_64/libc.a(vsnprintf.o): in function `vsnprintf':
vsnprintf.c:(.text.vsnprintf+0x4c): warning: warning: the printf functions add several kilobytes of bloat.
/usr/lib64/gcc/x86_64-pc-linux-gnu/9.3.0/../../../../x86_64-pc-linux-gnu/bin/ld: CMakeFiles/test_suite_x509write.dir/test_suite_x509write.c.o: in function `test_x509_crt_check':
test_suite_x509write.c:(.text+0x976): undefined reference to `mbedtls_pk_setup_rsa_alt'
collect2: error: ld returned 1 exit status

The bloat warning message comes from my special libc. You can disregard it here.

**Steps to reproduce**  

----------------------------------------------------------------
## Enhancement\Feature Request

**Justification - why does the library need this feature?**  

**Suggested enhancement**  

-----------------------------------------------------------------

## Question

**Please first check for answers in the [Mbed TLS knowledge Base](https://tls.mbed.org/kb). If you can't find the answer you're looking for then please use the [Mbed TLS mailing list](https://lists.trustedfirmware.org/mailman/listinfo/mbed-tls)**

--- a/ChangeLog.d/fix-rsaalt-test-guards.txt
+++ b/ChangeLog.d/fix-rsaalt-test-guards.txt
@@ -1,0 +1,3 @@
+Bugfix
+   * Fix compile error where MBEDTLS_RSA_C and MBEDTLS_X509_CRT_WRITE_C are
+     defined, but MBEDTLS_PK_RSA_ALT_SUPPORT is not defined. Fixes #3174.

--- a/tests/scripts/all.sh
+++ b/tests/scripts/all.sh
@@ -1948,6 +1948,18 @@ component_test_depends_py_pkalgs_psa () {
     tests/scripts/depends.py pkalgs
 }
 
+component_build_no_pk_rsa_alt_support () {
+    msg "build: !MBEDTLS_PK_RSA_ALT_SUPPORT" # ~30s
+
+    scripts/config.py full
+    scripts/config.py unset MBEDTLS_PK_RSA_ALT_SUPPORT
+    scripts/config.py set MBEDTLS_RSA_C
+    scripts/config.py set MBEDTLS_X509_CRT_WRITE_C
+
+    # Only compile - this is primarily to test for compile issues
+    make CC=gcc CFLAGS='-Werror -Wall -Wextra -I../tests/include/alt-dummy'
+}
+
 component_build_module_alt () {
     msg "build: MBEDTLS_XXX_ALT" # ~30s
     scripts/config.py full

--- a/tests/suites/test_suite_x509write.function
+++ b/tests/suites/test_suite_x509write.function
@@ -345,7 +345,7 @@ void x509_crt_check(char *subject_key_file, char *subject_pwd,
 
     issuer_key_type = mbedtls_pk_get_type(&issuer_key);
 
-#if defined(MBEDTLS_RSA_C)
+#if defined(MBEDTLS_RSA_C) && defined(MBEDTLS_PK_RSA_ALT_SUPPORT)
     /* For RSA PK contexts, create a copy as an alternative RSA context. */
     if (pk_wrap == 1 && issuer_key_type == MBEDTLS_PK_RSA) {
         TEST_ASSERT(mbedtls_pk_setup_rsa_alt(&issuer_key_alt,

